### PR TITLE
feat: extend toColor function by design tokens, update storybook and tests

### PR DESCRIPTION
**Pull request by pbochnak-sage**
### Proposed behaviour
Extend toColor function, so that the user can get chosen color from design tokens, providing "--tokenName" pattern as a prop. 
Update storybook, add usage examples.

### Current behaviour
toColor function lets users override color variant, providing, any color from the palette or any valid css color value.

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [x] All themes are supported if required
- [x] Unit tests added or updated if required
- [ ] Cypress automation tests added or updated if required
- [x] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required

#### QA

- [ ] Tested in CodeSandbox/storybook
- [ ] Add new Cypress test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions

The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.

### CODE REVIEW

**Review by jhajduk-sage — DISMISSED** [review]

File: docs/colors.stories.mdx [hunk 1]
```diff
@@ -33,15 +33,29 @@ Using the name of a palette color ensures if we update the theme your components
...
 <Typography color='whiteOpacity22'>...</Typography>
-```
\ No newline at end of file
+```
+
+## Design tokens
+
+Using design tokens ensures visual consistency across the application. If the designers update colors using a design tool, we have got always updated values.
+
+The name of the chosen design token must be always preceded by `--`, for example, if we want to use `colorsYang100` token in our code, we have to pass `--colorsYang100` as a prop. See [Design Tokens](https://zeroheight.com/2ccf2b601/p/217e24-design-tokens/b/46fb17) documentation for more information.
```
Comment (mkrds) [inline docs/colors.stories.mdx:55]: This link leads to a doc page that is protected by a password. Not sure if it's necessary here tbh

File: src/components/flat-table/flat-table-cell/flat-table-cell.component.js [hunk 2]
```diff
@@ -122,7 +122,7 @@ FlatTableCell.propTypes = {
   reportCellWidth: PropTypes.func,
   /** Sets a custom vertical right border */
   verticalBorder: PropTypes.oneOf(["small", "medium", "large"]),
-  /** Sets a custom vertical right border */
+  /** Sets a custom vertical right border, provide design token, any color from palette or any valid css color value. */
```
Comment (mkrds) [inline src/components/flat-table/flat-table-cell/flat-table-cell.component.js:125]:
```suggestion
  /** Sets a vertical right border color, provide design token, any color from palette or any valid css color value. */
```

File: src/components/flat-table/flat-table-row-header/flat-table-row-header.component.js [hunk 3]
```diff
@@ -86,7 +86,7 @@ FlatTableRowHeader.propTypes = {
   onKeyDown: PropTypes.func,
   /** Sets a custom vertical right border */
   verticalBorder: PropTypes.oneOf(["small", "medium", "large"]),
-  /** Sets a custom vertical right border */
+  /** Sets a custom vertical right border, provide design token, any color from palette or any valid css color value. */
```
Comment (mkrds) [inline src/components/flat-table/flat-table-row-header/flat-table-row-header.component.js:89]:
```suggestion
  /** Sets a vertical right border color, provide design token, any color from palette or any valid css color value. */
```

File: docs/colors.stories.mdx [hunk 4]
```diff
@@ -33,15 +33,29 @@ Using the name of a palette color ensures if we update the theme your components
...
 <Typography color='brilliantGreenShade30'>...</Typography>
 <Typography color='blackOpacity55'>...</Typography>
 <Typography color='whiteOpacity22'>...</Typography>
-```
\ No newline at end of file
+```
+
+## Design tokens
+
+Using design tokens ensures visual consistency across the application. If the designers update colors using a design tool, we have got always updated values.
```
Comment (mkrds) [inline docs/colors.stories.mdx:53]:
```suggestion
Using design tokens ensures visual consistency across the application and means that we will always have up-to-date values, if the colors are updated by designers using design tools
```

File: src/style/utils/get-color-value.js [hunk 5]
```diff
@@ -0,0 +1,8 @@
+import tokens from "@sage/design-tokens/js/base/common";
```
Comment (mkrds) [inline src/style/utils/get-color-value.js:1]: Would be nice if this helper would be written in TypeScript as we're making the transition now. You don't have to rewrite everything else it's just that we have a rule that anything that is new is meant to be written in TS

**Review by samtjo — COMMENTED** [review]
Nothing to add on top of Maciek's comments, happy to approve once they have been resolved 👍🏻

**Review by mkrds — DISMISSED** [review]

**Review by samtjo — APPROVED** [review]

**Review by DlgSHi — APPROVED** [review]

**Review by mkrds — APPROVED** [review]

**Review by samtjo — APPROVED** [review]

**Pull request merged**